src/drivers/sound/sound-sdl.c

```diff
@@ -222,11 +215,112 @@ void snd_sdl_callback (void *user, Uint8 *buf, int cnt)
 	}
 }
 
+/*
+ * Resampling can handle only input_freq / output_freq in range [0.5 1.0]
+ */
+// lowpass filter cut off, use 0 to remove lowpass filter
+#define CUT_OFF 6000
+
+static
+void sdl_snd_fix_lowpass (sound_sdl_t *drv, int chn, int freq, int srate)
+{
+	unsigned i;
+
+	for (i = 0; i < chn; i++) {
+		snd_iir2_set_lowpass (
+			&drv->sdl_lowpass_iir2[i],
+			freq, srate
+		);
+
+		snd_iir2_reset (&drv->sdl_lowpass_iir2[i]);
+	}
+}
+
+void snd_sdl_callback_resample (sound_sdl_t *drv, uint16_t *dest, int cnt)
+{
+	int             n;
+	sound_sdl_buf_t *src;
+
+	cnt /= 2;
+
+	int feed = 0;
+	int old_cnt = cnt;
+	uint16_t * old_dest = dest;
+
+	for ( ; cnt > 0 ; cnt--) {
+		src = drv->head;
+		sound_sdl_buf_t *old_src = src;
+		if (src->d_idx >= src->cnt/2) {
+
+			drv->head = src->next;
+
+			if (drv->head == NULL) {
+				drv->tail = NULL;
+			}
+
+			src->next = drv->free;
+			drv->free = src;
+
+			src = drv->head;
+
+			if (!src) {
+#if DEBUG_SND_SDL >= 1
+				fprintf (stderr, "snd-sdl: buffer underrun\n");
+#endif
+				emscripten_log(EM_LOG_CONSOLE, "snd-sdl: buffer underrun");
+				memset (dest, 0, cnt*2);
+				return;
+			}
+			src->d_idx = old_src->d_idx - (int)old_src->d_idx;
```

Comment:
I'm unsure I get that right wut hte whole things work works flawless with Hampa Hug code